### PR TITLE
feat: shell tab completion instructions

### PR DESCRIPTION
**Pull request by algchoo**
### Proposed Change

added instructions for setting up tab completion for bash/zsh on linux/macos

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -183,6 +183,45 @@ Copy the output and run the command on the system your wish to install the colle
...
+
+To setup zsh completion for bindplanectl on MacOS:
+1. Include the following lines in `~/.zshrc`&nbsp;
+```
+autoload -Uz compinit
+compinit
+```
+2. Locate `fpath` by running `echo $fpath`, there may be several listed, some may not exist, use an existing one in the next step.
+3. Run the following command to generate the zsh tab completion script.\
+`bindplanectl completion --zsh ><YOUR FPATH HERE>/_bindplanectl`
```
Comment (jsirianni) [inline docs/install.md:222]: zsh is a sub command, not a flag
```suggestion
`bindplanectl completion zsh ><YOUR FPATH HERE>/_bindplanectl`
```

**Review by jsirianni — CHANGES_REQUESTED** [review]

**Comment by dsvanlani** [comment]
I was not able to get completion working with these instructions on Mac and zsh.

**Comment by jsirianni** [comment]
I was able to get this working on Linux (zsh) but not macOS. I think it is working generally, but we have some consistency issues between macOS installs.

File: docs/install.md [hunk 2]
```diff
@@ -183,6 +183,45 @@ Copy the output and run the command on the system your wish to install the colle
 
 After running the collector install command, the collector will appear in the `bindplanectl get agents` output.
 
+### Shell Completion
+
+#### Linux: bash
+
+1. Verify that `bash-completion` is installed on the host
+2. `sudo su` Switching to root
+3. `bindplane completion bash >/etc/bash_completion.d/bindplane` appends the output to a file in the bash completion directory
```
Comment (jsirianni) [inline docs/install.md:192]: We should use `sudo tee` when redirecting output to a file owned by root, when not running as root.

Comment (jsirianni) [inline docs/install.md:192]: Lets always use bindplanectl when running commands, bindplane (binary) may not support cli in the future.

```suggestion
3. `bindplanectl completion bash >/etc/bash_completion.d/bindplanectl` appends the output to a file in the bash completion directory
```

File: docs/install.md [hunk 3]
```diff
@@ -183,6 +183,45 @@ Copy the output and run the command on the system your wish to install the colle
...
+4. Restart the shell
+
+#### macOS: bash
```
Comment (jsirianni) [inline docs/install.md:195]: I am okay with removing this for now
1) I cant get it to work
2) Most macOS users will be using zsh

**Review by jsirianni — CHANGES_REQUESTED** [review]

File: docs/install.md (hunk 3 above)
Comment (algchoo) [reply docs/install.md]: removed this portion of the instructions

**Review by jsirianni — APPROVED** [review]

**Comment by andykellr** [comment]
It would be nice to add an .oh-my-zsh plugin, like https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/gh/gh.plugin.zsh

**Comment by andykellr** [comment]
I'm good with merging this, but this is what I did:

1. Verify that compinit is already in .zshrc

2.
```sh
compinit; echo $fpath # /Users/andy/.oh-my-zsh/completions in the list but doesn't exist
mkdir /Users/andy/.oh-my-zsh/completions
bindplanectl completion zsh >/Users/andy/.oh-my-zsh/completions/_bindplanectl
```

3. Restart zsh

**Review by dsvanlani — APPROVED** [review]

**Pull request merged**